andela/dahlia-ah-backend
# #167165006 Role Based Access Control

#### Pivotal tracker story
[#167165006](https://www.pivotaltracker.com/story/show/167165006)
#### What does this PR do?
Control user access based on role
#### Summary of Task
- Create role model
- Create function to control users access
- Add test
- Add swagger documentation
#### How can this be tested?
- Use **git clone** to clone this repository to local branch
- Use **npm install** to download all necessary packages
- Ensure your database is created or you can use **sequelize db:create** if database does not exist
- Use **npm run test**  to test
- Go your POSTMAN and try to access a route with a different access priviledge
#### Screenshots (if appropriate)
![Role based access control](https://user-images.githubusercontent.com/7049701/63071282-974b7000-bf16-11e9-8702-4fee8d379137.png)
Flow Chart of time to read a novel

<img width="658" alt="Screenshot 2019-08-15 at 4 30 15 AM" src="https://user-images.githubusercontent.com/7049701/63071304-a6322280-bf16-11e9-92a1-56be7ec62633.png">
Screenshot of an unaccessible novel page by reader
